documentation/typescript/ts-handle-session-expiration.asciidoc

```diff
@@ -0,0 +1,82 @@
+---
+title: "Appendix: Handle session expiration in TypeScript"
+order: 141
+layout: page
+---
+
+= Appendix: Handle session expiration in TypeScript
+
+Vaadin provides a built-in <<client-middleware#, middleware>> `InvalidSessionMiddleWare` that you can use to detect session expiration, for example, to show a login view to the user.
+== How to use the InvalidSessionMiddleWare?
+
+The `InvalidSessionMiddleWare` requires a `OnInvalidSessionCallback` as a constructor parameter. The `OnInvalidSessionCallback` is a function that takes no parameters and should return a promise of `LoginResult`. `LoginResult` contains the metadata of a login result, includes:
```

Comment:
The third sentence, starting with "`LoginResult` contains the metadata [...]" is grammatically incorrect. Either 
> `LoginResult` contains the metadata of a login result, including:
or 
> `LoginResult` contains the metadata of a login result; it includes:

Reply:
done